bin/index.js

```diff
@@ -11,44 +11,77 @@ const jsonFormatter = require('../lib/formatters/json');
 const yamlFormatter = require('../lib/formatters/yaml');
 const SVG = require('../lib/svg');
 
-// Declare the app
-const app = (filePath, options) => {
-  fs.readFile(filePath, (err, data) => {
-    if (err) {
-      switch (err.code) {
-        case 'ENOENT':
-          console.error(
-            `${colors.red('[Error]')} There file ${filePath} doesn't exist`);
-          break;
-        default:
-          console.error(
-            `${colors.red('[Error]')} There was an error opening the ` +
-            `file: ${err.message}`);
+const parseStdin = () => {
+  return new Promise((resolve, reject) => {
+    let data = '';
+    
+    process.stdin.on('data', (chunk) => {
+      data += chunk;
+    });
+
+    process.stdin.on('error', (error) => {
+      reject(`There was an error parsing the STDIN data: ${error}`);
+    });
+
+    process.stdin.on('end', function() {
+      resolve(data);
+    });
+  });
+};
+
+const parseFile = (filePath) => {
+  return new Promise((resolve, reject) => {
+    fs.readFile(filePath, (err, data) => {
```

Comment:
I detected another small corner case in your solution. If you execute the tool without any argument nor `stdin` data, the app crash. We should check here if `filePath` is not `undefined` before try to read the file. If not, we should reject the Promise.

Reply:
Good catch. Since I'm having to call app manually at the end due to how commander handles the edge case of calling the command with no arguments, I forgot to handle the case where there was legitimately no input. I think a better solution is to catch that before we get to parseFile and print the help information instead.

Reply:
I do have one lingering concern though. Right now it's no longer clear that you can pass in stdin instead from the generated help information. My first thought is that I could change `[file]` to `<file|stdin>` in the `arguments` call, but I'm open to suggestions

Reply:
You're right. I agree it's better to display the help of the tool instead of an error. About how to show that you can use `file` or `stdin`, maybe we can use `[file|stdin]`. I check how `jq` manages this issue, but it doesn't provide any information about `stdin`. So, I think that `[file|stdin]` is fine. What do you think?

Reply:
Looks good to me